text/0000-debug-noop-in-release-builds.md

```diff
@@ -0,0 +1,30 @@
+- Feature Name: debug-noop-in-release
+- Start Date: 2014-03-04
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+The `{:?}` format specifier should not print values in release builds.
+
+# Motivation
+
+Many commonly used methods, such as `unwrap`, use `{:?}` to print out representations of values if their conditions are not met. This is useful for debugging but is not so useful in release builds, and it can bloat up the generated binaries. Worse, it adds opaque calls (as `Debug` is frequently a lot of code and rarely gets inlined), which can mess up the inlining heuristics and cause cascades of missed optimization opportunities.
+
+# Detailed design
+
+In release mode, `{:?}` should simply print `(debugging representation omitted in release builds)`. This makes it clear to the programmer what should be done to reenable the feature. If debugging information is *really* desired in release builds, the method from the `Debug` trait can still be called explicitly.
```

Comment:
If we want to implement this behaviour for the explicit methods, then we can't make the Debug impls `#[cfg(debug]]` (which it sounds like you want from the code bloat motivation), otherwise the only code bloat you elide is due to inlining. If that is the case, not using `{:?}` seems like a better solution.